### ISSUE TITLE
Is it possible to mock `process.cwd()`?

### ISSUE THREAD

**Issue by tommy-mitchell** [iambumblehead/esmock#196]
I'm trying to mock `process.cwd()` to return the temporary directory that my tests are running in:

```js
const someTestSetupFn = async (tempDir) => {
	// has direct and in-direct calls to `process.cwd()`
	const thingBeingTested = await esmock('my-path', {}, {
		'node:process': {cwd: () => tempDir},
	});

	// ... do test stuff
};
```

However, I'm getting this error:

```
TypeError: Cannot assign to read only property 'version' of object '#<process>'
```

<details>
<summary>Stack</summary>

```
at file:///.../node_modules/esmock/src/esmock.js:2:4241
at Array.reduce (<anonymous>)
at at (file:///.../node_modules/esmock/src/esmock.js:2:4222)
at lt (file:///.../node_modules/esmock/src/esmock.js:2:4292)
at ut (file:///.../node_modules/esmock/src/esmock.js:2:4349)
at kt (file:///.../node_modules/esmock/src/esmock.js:2:5158)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async N (file:///.../node_modules/esmock/src/esmock.js:2:5593)
at async gt (file:///.../node_modules/esmock/src/esmock.js:2:5935)
at async file:///.../node_modules/esmock/src/esmock.js:3:155
```

</details>

**Comment by iambumblehead** [comment]
@tommy-mitchell it _should_ be possible. Let's try adding a test to esmock and taking a quick look...

**Comment by iambumblehead** [comment]
related MR: https://github.com/iambumblehead/esmock/pull/197

Whats happening looks basically like this,
```javascript
const nodeProcess = await import('node:process')
const mock = Object.create(nodeProcess.default)

Object.keys(nodeProcess.default).reduce((mockFinal, key) => {
  //  TypeError: "Cannot assign to read only property 'version' of object '#<process>'"
  mockFinal[key] = nodeProcess.default[key]
  return mockFinal
}, mock)
```

Object.create should give us our own object (different from nodeProcess.default) on which properties can be written without affecting objects inherited at the prototype chain. I'm not sure how best to resolve this atm as the current flow is needed in order for other tests to pass.

**Comment by iambumblehead** [comment]
Object.defineProperty can be used to get around this limitation... I think a solution will be ready soon

**Comment by iambumblehead** [comment]
@tommy-mitchell if you're around an wish to review the MR that's great. If you don't respond, I'll plan to merge and publish solution in order to expedite things for you. Thanks for opening this issue.

**Comment by tommy-mitchell** [comment]
I'll take a look :)

**Comment by iambumblehead** [comment]
@tommy-mitchell the change is published to 2.2.1 release https://www.npmjs.com/package/esmock
